server/api/core/core.js

```diff
@@ -364,29 +363,93 @@ export default {
     return null;
   },
 
+  /**
+   * @name connectionDataStore
+   * @summary a construct to store per-connection data. Once a connection is
+   * closed, the data is lost. This is server-side companion to Sessions in the
+   * client. connectionDataStore has two methods: get and set
+   * @memberof Core
+   */
+  connectionDataStore: (function () {
+    const inMemoryCache = {};
+    const connectionKey = "connection-data";
+
+    function connectionData() {
+      // similar to Meteor.userId()
+      const invocation =
+        DDP._CurrentMethodInvocation.get() ||
+        DDP._CurrentPublicationInvocation.get();
+
+      if (!invocation) {
+        // operating outside of a connection
+        return inMemoryCache;
+      }
+
+      const { connection } = invocation;
+
+      if (!(connectionKey in connection)) {
+        connection[connectionKey] = {};
+      }
+
+      return connection[connectionKey];
+    }
+
+    return {
+      get(key) {
+        return connectionData()[key];
+      },
+
+      set(key, val) {
+        connectionData()[key] = val;
+      }
+    };
+  })(),
+
+  debuggingQueryCount: 1,
+  debuggingSavedQueryCount: 1,
   /**
    * @name getShopId
    * @method
    * @memberof Core
-   * @summary Get shop ID
-   * @todo This should intelligently find the correct default shop Probably whatever the main shop is or marketplace
-   * @param  {String} userId User ID String
+   * @summary Get shop ID, first by checking the current user's preferences
+   * then by getting the shop by the current domain.
+   * @todo should we return the Primary Shop if none found?
    * @return {StringId}        active shop ID
    */
-  getShopId(userId) {
-    check(userId, Match.Maybe(String));
+  getShopId() {
```

Comment:
^ this is the breaking change.
In order to simplify things a bit, I removed the `userId` parameter/feature. There were only a handful of uses of that feature, which have been updated in this PR. Of course, I could not account for all of the plugins that exist in the world, so your call on whether to keep this change.